### ISSUE TITLE
NGINX ingress controller pod missing tcp-services/udp-services args

### ISSUE THREAD

**Issue by peyoz** [rancher/rke#2741]
**RKE version:**  v1.3.2

**Docker version: (`docker version`,`docker info` preferred)**
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.6.3-docker)
  scan: Docker Scan (Docker Inc., v0.9.0)

Server:
 Containers: 105
  Running: 88
  Paused: 0
  Stopped: 17
 Images: 80
 Server Version: 20.10.10
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 5b46e404f6b9f661a205e28d59c982d3634148f8
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1160.45.1.el7.x86_64
 Operating System: Red Hat Enterprise Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 23.54GiB
 Name: itdgtosax000266
 ID: KRCO:UW3T:SE5B:Q3IQ:DZZN:LGG4:DQIN:OR3Q:I7U2:KMBB:LFSK:HFRX
 Docker Root Dir: /data/docker
 Debug Mode: false
 HTTP Proxy: **Redacted**
 HTTPS Proxy: **Redacted**
 No Proxy: **Redacted**
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
NAME="Red Hat Enterprise Linux Server"
VERSION="7.9 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.9:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.9"

3.10.0-1160.45.1.el7.x86_64

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)** bare-metal

**cluster.yml file:**
```
ingress:
    provider: nginx
    node_selector:
      app: ingress
```

**Steps to Reproduce:** Upgrade to kubernetes_version: v1.19.16-rancher1-1 and check nginx pod running config

**Results:**
```
Containers:
  controller:
    Container ID:  docker://555328dafe463c8c46bfa129d2f94afb3d1e22497331a52e35a86b1e3f66386a
    Image:         rancher/nginx-ingress-controller:nginx-0.49.3-rancher1
    Image ID:      docker-pullable://rancher/nginx-ingress-controller@sha256:1617a3df09627d8b3b9b9d54c4791637dbbc6c75ec213f9d2ec886d82b03b48c
    Ports:         80/TCP, 443/TCP, 8443/TCP
    Host Ports:    80/TCP, 443/TCP, 8443/TCP
    Args:
      /nginx-ingress-controller
      --default-backend-service=$(POD_NAMESPACE)/default-http-backend
      --election-id=ingress-controller-leader
      --ingress-class=nginx
      --configmap=$(POD_NAMESPACE)/ingress-nginx-controller
      --validating-webhook=:8443
      --validating-webhook-certificate=/usr/local/certificates/cert
      --validating-webhook-key=/usr/local/certificates/key
```

**Comment by superseb** [comment]
We follow upstream templates, if they are removed there you will need to provide them using `extra_args` yourself. Let me know if this doesn't work.

**Comment by peyoz** [comment]
Looks like this problem was found and fixed here #35943
